resources.tf

```diff
@@ -0,0 +1,315 @@
+#providers
+provider "aws" {
+	shared_credentials_file  = "C:\\Users\\ThulsiDossK\\.aws\\credentials"
+	region = "${var.region}"
+}
+
+#resources
+resource "aws_vpc" "vpc" {
+  cidr_block = "${var.cidr_vpc}"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+  tags = {
+    "Environment" = "${var.environment_tag}"
+  }
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  tags = {
+    "Environment" = "${var.environment_tag}"
+  }
+}
+
+resource "aws_eip" "ip" {
+ vpc =  true
+}
+resource "aws_nat_gateway" "natgw1" {
+ allocation_id="${aws_eip.ip.id}"
+  subnet_id = "${aws_subnet.subnet_public.id}"
+  tags = {
+    "Environment" = "${var.environment_tag}"
+  }
+}
+
+resource "aws_subnet" "subnet_public" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  cidr_block = "${var.cidr_subnet}"
+  map_public_ip_on_launch = "true"
+  availability_zone = "${var.availability_zone}"
```

Comment:
It is possible to deploy a VPC without having to pass `availability_zone` as a variable. I will ask you to think about options and discuss possible solutions with me in the chat